<launch>
  <!-- master DAVIS -->
  <node ns="davis_left" name="davis_ros_driver" pkg="davis_ros_driver" type="davis_ros_driver" output="screen">
    <param name="serial_number" type="str" value="00000306" />
    <rosparam command="load" file="$(find davis_ros_driver)/config/DAVIS240C.yaml" />
    <param name="master" value="True"/>
    <!-- timeout before resetting time -->
    <param name="reset_timestamps_delay" value="2.0"/>
  </node>

  <!-- slave DAVIS -->
  <node ns="davis_right" name="davis_ros_driver" pkg="davis_ros_driver" type="davis_ros_driver" output="screen">
    <param name="serial_number" type="str" value="00000311" />
    <rosparam command="load" file="$(find davis_ros_driver)/config/DAVIS240C.yaml" />
    <param name="master" value="False"/>
    <!-- get reset signal from master -->
    <remap from="reset_timestamps" to="/davis_left/reset_timestamps"/>
  </node>

  <!-- visualization -->
  <node name="dvs_renderer_left" pkg="dvs_renderer" type="dvs_renderer">
    <param name="display_method" value="grayscale"/>
    <remap from="events" to="/davis_left/events" />
    <remap from="dvs_rendering" to="dvs_rendering_left" />
  </node>
  <node name="dvs_renderer_right" pkg="dvs_renderer" type="dvs_renderer">
    <param name="display_method" value="grayscale"/>
    <remap from="events" to="/davis_right/events" />
    <remap from="dvs_rendering" to="dvs_rendering_right" />
  </node>

  <!-- calib node -->
  <node name="dvs_calibration" pkg="dvs_calibration" type="stereo" output="screen">
    <remap from="camera_info_left" to="/davis_left/camera_info" />
    <remap from="events_left" to="/davis_left/events" />
    <remap from="camera_info_right" to="/davis_right/camera_info" />
    <remap from="events_right" to="/davis_right/events" />
    <remap from="set_camera_info_left" to="/davis_left/set_camera_info" />
    <remap from="set_camera_info_right" to="/davis_right/set_camera_info" />

    <param name="blinking_time_us" value="1000" />
    <param name="blinking_time_tolerance_us" value="500" />
    <param name="enough_transitions_threshold" value="100" />
    <param name="minimum_transitions_threshold" value="30" />
    <param name="minimum_transitions_threshold" value="30" />

    <param name="dots_w" value="5"/>
    <param name="dots_h" value="4"/>
    <param name="dot_distance" value="0.05"/>
  </node>
  
  <!-- display -->
  <node name="rqt_gui" pkg="rqt_gui" type="rqt_gui" args="--perspective-file $(find dvs_calibration_gui)/rqt/StereoCalibration.perspective" />

</launch>
